From f16ace03278d811e995ccd2a6a23a6d8ebe94eac Mon Sep 17 00:00:00 2001
From: Tommy Keswick <tkeswick@caltech.edu>
Date: Wed, 31 Jan 2018 22:02:58 +0000
Subject: [PATCH] Issue #355: fix PHP Notices on Organizations index page.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add isset() to fix PHP Notice: Undefined index: org_orderBy
    in …/organizations/index.php on line 153
Add isset() to fix PHP Notice: Undefined index: org_numberOfRecords
    in …/organizations/index.php on line 149
Add isset() to fix PHP Notice: Undefined index: org_pageStart
    in …/organizations/index.php on line 145
Add isset() to fix PHP Notice: Undefined index: org_startWith
    in …/organizations/index.php on line 140
Add isset() to fix PHP Notice: Undefined index: org_contactName
    in …/organizations/index.php on line 95
Add isset() to fix PHP Notice: Undefined index: org_contactName
    in …/organizations/index.php on line 94
Add isset() to fix PHP Notice: Undefined index: org_organizationRoleID
    in …/organizations/index.php on line 78
Add isset() to fix PHP Notice: Undefined index: org_organizationName
    in …/organizations/index.php on line 62
Add isset() to fix PHP Notice: Undefined index: org_organizationName
    in …/organizations/index.php on line 61
Set $reset variable to fix PHP Notice: Undefined variable: reset
    in …/organizations/index.php on line 58
Add isset() to fix PHP Notice: Undefined index: ref_script
    in …/organizations/index.php on line 32
---
 organizations/index.php | 23 +++++++++++++----------
 1 file changed, 13 insertions(+), 10 deletions(-)

diff --git a/organizations/index.php b/organizations/index.php
index 2c6641e6..67c7f30d 100644
--- a/organizations/index.php
+++ b/organizations/index.php
@@ -29,9 +29,12 @@
 //except we don't want it to retain if they press the 'index' button
 //check what referring script is
 
-if ($_SESSION['ref_script'] != "orgDetail.php"){
+if (isset($_SESSION['ref_script']) && $_SESSION['ref_script'] != "orgDetail.php") {
 	$reset='Y';
 }
+else {
+  $reset = 'N';
+}
 
 $_SESSION['ref_script']=$currentPage;
 
@@ -55,8 +58,8 @@
 	<tr>
 	<td class='searchRow'><label for='searchName'><b><?php echo _("Name (contains)");?></b></label>
 	<br />
-	<input type='text' name='searchOrganizationName' id='searchOrganizationName' style='width:145px' value="<?php if ($reset != 'Y') echo $_SESSION['org_organizationName']; ?>" /><br />
-	<div id='div_searchName' style='<?php if ((!$_SESSION['org_organizationName']) || ($reset == 'Y')) echo "display:none;"; ?>margin-left:123px;'><input type='button' name='btn_searchOrganizationName' value='<?php echo _("go!");?>' class='searchButton' /></div>
+	<input type='text' name='searchOrganizationName' id='searchOrganizationName' style='width:145px' value="<?php if ($reset != 'Y' && isset($_SESSION['org_organizationName'])) echo $_SESSION['org_organizationName']; ?>" /><br />
+	<div id='div_searchName' style='<?php if ((!isset($_SESSION['org_organizationName'])) || ($reset == 'Y')) echo "display:none;"; ?>margin-left:123px;'><input type='button' name='btn_searchOrganizationName' value='<?php echo _("go!");?>' class='searchButton' /></div>
 	</td>
 	</tr>
 
@@ -72,7 +75,7 @@
 		$organizationRole = new OrganizationRole();
 
 		foreach($organizationRole->allAsArray() as $display) {
-			if (($_SESSION['org_organizationRoleID'] == $display['organizationRoleID']) && ($reset != 'Y')){
+			if ((isset($_SESSION['org_organizationRoleID'])) && ($_SESSION['org_organizationRoleID'] == $display['organizationRoleID']) && ($reset != 'Y')) {
 				echo "<option value='" . $display['organizationRoleID'] . "' selected>" . $display['shortName'] . "</option>";
 			}else{
 				echo "<option value='" . $display['organizationRoleID'] . "'>" . $display['shortName'] . "</option>";
@@ -88,8 +91,8 @@
 	<tr>
 	<td class='searchRow'><label for='searchContact'><b><?php echo _("Contact Name (contains)");?></b></label>
 	<br />
-	<input type='text' name='searchContactName' id='searchContactName' style='width:145px' value="<?php if ($reset != 'Y') echo $_SESSION['org_contactName']; ?>" /><br />
-	<div id='div_searchContact' style='<?php if ((!$_SESSION['org_contactName']) || ($reset == 'Y')) echo "display:none;"; ?>margin-left:123px;'><input type='button' name='btn_searchContactName' value='<?php echo _("go!");?>' class='searchButton' /></div>
+	<input type='text' name='searchContactName' id='searchContactName' style='width:145px' value="<?php if ($reset != 'Y' && isset($_SESSION['org_contactName'])) echo $_SESSION['org_contactName']; ?>" /><br />
+	<div id='div_searchContact' style='<?php if ((!isset($_SESSION['org_contactName'])) || ($reset == 'Y')) echo "display:none;"; ?>margin-left:123px;'><input type='button' name='btn_searchContactName' value='<?php echo _("go!");?>' class='searchButton' /></div>
 	</td>
 	</tr>
 
@@ -134,20 +137,20 @@
 <?php
   //used to default to previously selected values when back button is pressed
   //if the startWith is defined set it so that it will default to the first letter picked
-  if (($_SESSION['org_startWith']) && ($reset != 'Y')){
+  if ((isset($_SESSION['org_startWith'])) && ($reset != 'Y')){
 	  echo "startWith = '" . $_SESSION['org_startWith'] . "';";
 	  echo "$(\"#span_letter_" . $_SESSION['org_startWith'] . "\").removeClass('searchLetter').addClass('searchLetterSelected');";
   }
 
-  if (($_SESSION['org_pageStart']) && ($reset != 'Y')){
+  if ((isset($_SESSION['org_pageStart'])) && ($reset != 'Y')){
 	  echo "pageStart = '" . $_SESSION['org_pageStart'] . "';";
   }
 
-  if (($_SESSION['org_numberOfRecords']) && ($reset != 'Y')){
+  if ((isset($_SESSION['org_numberOfRecords'])) && ($reset != 'Y')){
 	  echo "numberOfRecords = '" . $_SESSION['org_numberOfRecords'] . "';";
   }
 
-  if (($_SESSION['org_orderBy']) && ($reset != 'Y')){
+  if ((isset($_SESSION['org_orderBy'])) && ($reset != 'Y')){
 	  echo "orderBy = \"" . $_SESSION['org_orderBy'] . "\";";
   }
 
